Common/src/main/java/com/blamejared/crafttweaker/api/action/base/IAction.java

```diff
@@ -49,6 +49,23 @@ public interface IAction {
      */
     String describe();
     
+    /**
+     * Gets the name of the system that is responsible for the execution of the action.
+     *
+     * <p>The system usually corresponds to the name of the mod that created the action and is thus attempting to apply
+     * it. For example, adding a recipe is done through CraftTweaker, so the system name is {@code "CraftTweaker"}. Mods
+     * that have various components or would like to separate various integrations from each other can return more
+     * complex strings.</p>
+     *
+     * <p>Note that the system name will be used for logging, therefore it is suggested to avoid overly verbose names or
+     * acronyms that are not widely known. Returning {@code null} or an empty name is disallowed.</p>
+     *
+     * @return The name of the system that is responsible for this action.
+     *
+     * @since 10.1.0
+     */
+    String systemName();
+    
```

Comment:
>  The system usually corresponds to the name of the mod that created the action and is thus attempting to apply it. For example, adding a recipe is done through CraftTweaker, so the system name is {@ code "CraftTweaker"}.

In Ice and Fire Tweaker, I have custom actions that add recieps to their recipe registries, should I be returning "CraftTweaker" or "IceAndFireTweaker"?

Reply:
IceAndFireTweaker, I'll try to make it clearer in the Javadoc